numpy/core/tests/test_umath_complex.py

```diff
@@ -107,18 +107,18 @@ def _check_inf_nan(dummy):
                 if not np.isinf(z.real) or not np.isnan(z.imag):
```

Comment:
Really beyond the PR, but here and below, I think the following is clearer (no "double negatives")
```
assert np.isinf(z.real) and np.isnan(z.imag), msgform % (z.real, z.imag)
```

Reply:
Yep, the `test_umath_complex.py` module could use a major refactoring. I sort of started down that road before stopping myself :)

Reply:
Note that the functions were created so that they could be used with the previous `yield` statement.